### DISCUSSION TITLE
The closed #230, but I've figured-out why it happens.

### DISCUSSION THREAD

**Discussion by Eirikur** [agronholm/apscheduler#581] [General]
The #230 issue is that add_job() generates a 'Too many arguments' error, when the parameter list to add_job is perfectly fine.

The issue, which may not be fixable, really, or appropriate to fix, though it seems desirable, is that this is a mismatch between the user-supplied function and the user-supplied arguments.  I changed the function I was supplying and not the arguments and had this same problem. Unfortunately, most kinds of callback system can easily have this problem.

APScheduler is a real pleasure to work with, BTW.

**Comment by agronholm** [comment]
If you supply more arguments than the target function can take, that's never going to work, and the only fix is to change either the list of of arguments passed to `add_job()` or the function definition. It's definitely not an APScheduler problem, but a user error.

> APScheduler is a real pleasure to work with, BTW.

Thanks!
